File: packages/react-native-bottom-tabs/android/src/main/java/com/rcttabview/events/OnNativeLayoutEvent.kt
package com.rcttabview.events

import com.facebook.react.bridge.Arguments
import com.facebook.react.uimanager.events.Event
import com.facebook.react.uimanager.events.RCTEventEmitter

class OnNativeLayoutEvent(viewTag: Int, private val width: Double, private val height: Double) :
  Event<TabLongPressEvent>(viewTag) {

  companion object {
    const val EVENT_NAME = "onNativeLayout"
  }

  override fun getEventName(): String {
    return EVENT_NAME
  }

  override fun dispatch(rctEventEmitter: RCTEventEmitter) {
    val event = Arguments.createMap().apply {
      putDouble("width", width)
      putDouble("height", height)
    }
    rctEventEmitter.receiveEvent(viewTag, eventName, event)
  }
}

File: packages/react-native-bottom-tabs/ios/Events/OnNativeLayoutEvent.swift
import React

@objcMembers
public class OnNativeLayoutEvent: NSObject, RCTEvent {
  private var size: CGSize
  public var viewTag: NSNumber

  public var eventName: String {
    return "onNativeLayout"
  }

  public init(reactTag: NSNumber, size: CGSize) {
    self.viewTag = reactTag
    self.size = size
    super.init()
  }

  public class func moduleDotMethod() -> String {
    return "RCTEventEmitter.receiveEvent"
  }
  
  public func canCoalesce() -> Bool {
      return false
  }

  public func arguments() -> [Any] {
    return [
      viewTag,
      RCTNormalizeInputEventName(eventName) ?? eventName,
      [
        "width": size.width,
        "height": size.height
      ]
    ]
  }
}
